internal/services/realtimeengine/containersrealtime/containers-realtime.go

```diff
@@ -0,0 +1,199 @@
+	}
+
+	return result, nil
+}
+
+// parseContainersFile parses the containers file and returns a list of images.
+func parseContainersFile(filePath string) ([]types.ImageModel, error) {
+	extractor := imagesExtractor.NewImagesExtractor()
+
+	// Extract files from the scan path (directory containing the file)
+	scanPath := filepath.Dir(filePath)
+
+	logger.PrintfIfVerbose("Scanning directory for container images: %s", scanPath)
+
+	// Ensure the directory exists
+	if _, statErr := os.Stat(scanPath); os.IsNotExist(statErr) {
+		return nil, errors.Errorf("directory does not exist: %s", scanPath)
+	}
+
+	files, envVars, _, err := extractor.ExtractFiles(scanPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "error extracting files")
+	}
+
+	images, err := extractor.ExtractAndMergeImagesFromFilesWithLineInfo(files, []types.ImageModel{}, envVars)
+	if err != nil {
+		return nil, errors.Wrap(err, "error merging images")
+	}
+
+	logger.PrintfIfVerbose("Extracted %d container images", len(images))
+	return images, nil
+}
+
+// scanImages scans the extracted images using the realtime scanner.
+func (c *ContainersRealtimeService) scanImages(images []types.ImageModel, filePath string) (results *ContainerImageResults, err error) {
+	logger.PrintfIfVerbose("Scanning %d images for vulnerabilities", len(images))
+
+	var requestImages []wrappers.ContainerImageRequestItem
+	for _, img := range images {
+		imageName, imageTag := splitToImageAndTag(img.Name)
+
+		logger.PrintfIfVerbose("Processing image: %s:%s", imageName, imageTag)
+
+		requestImages = append(requestImages, wrappers.ContainerImageRequestItem{
+			ImageName: imageName,
+			ImageTag:  imageTag,
+		})
+	}
+
+	request := &wrappers.ContainerImageRequest{
+		Images: requestImages,
+	}
+
+	response, err := c.RealtimeScannerWrapper.ScanImages(request)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.PrintfIfVerbose("Received scan results for %d images", len(response.Images))
+
+	result := c.buildContainerImageResults(response.Images, images, filePath)
+	return &result, nil
+}
+
+// buildContainerImageResults builds ContainerImageResults from response and images
+func (c *ContainersRealtimeService) buildContainerImageResults(responseImages []wrappers.ContainerImageResponseItem, images []types.ImageModel, filePath string) ContainerImageResults {
+	var result ContainerImageResults
+	for i, respImg := range responseImages {
+		var locations []realtimeengine.Location
+		if i < len(images) {
+			locations = convertLocations(images[i].ImageLocations)
+		}
+
+		containerImage := ContainerImage{
+			ImageName:       respImg.ImageName,
+			ImageTag:        respImg.ImageTag,
+			FilePath:        filePath,
+			Locations:       locations,
+			Status:          respImg.Status,
+			Vulnerabilities: convertVulnerabilities(respImg.Vulnerabilities),
+		}
+		result.Images = append(result.Images, containerImage)
+	}
+	return result
```

Comment:
more efficient way is to preallocating the size for the result if you know what it will be:
```suggestion
	result := make(ContainerImageResults, 0, len(responseImages)
	for i, respImg := range responseImages {
		var locations []realtimeengine.Location
		if i < len(images) {
			locations = convertLocations(images[i].ImageLocations)
		}

		containerImage := ContainerImage{
			ImageName:       respImg.ImageName,
			ImageTag:        respImg.ImageTag,
			FilePath:        filePath,
			Locations:       locations,
			Status:          respImg.Status,
			Vulnerabilities: convertVulnerabilities(respImg.Vulnerabilities),
		}
		result.Images[i] = containerImage
	}
	return result
```